### PR TITLE
tests: Bluetooth: Mesh: Add some improvements to mesh BabbleSim tests

### PR DESCRIPTION
**Pull request by PavelVPV**
This PR adds the following improvements:
- If segmented message transmission fails, the error code is passed back to caller. This helps in debugging test failure;
- If message reception fails, the error code is passed back to caller. This helps in debugging test failure;
- In provisioning tests prevents the provisioner from provisioning a device before the provisioner is configured.

### CODE REVIEW

File: tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_persistence.c [hunk 1]
```diff
@@ -265,6 +267,10 @@ static void unprovisioned_beacon(uint8_t uuid[16], bt_mesh_prov_oob_info_t oob_i
 {
 	static bool once;
 
+	if (!provisioner_ready) {
```
Comment (alxelax) [inline tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_persistence.c:270]: The same here. It'd be well to see the reason why the provisioner cannot start. Any information message is welcome.

Comment (alxelax) [inline tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_persistence.c:270]: The same here. It'd be well to see the reason why the provisioner cannot start. Any information message is welcome.

**Review by alxelax — COMMENTED** [review]
LGTM just minor comments

File: tests/bluetooth/bsim_bt/bsim_test_mesh/src/mesh_test.c [hunk 2]
```diff
@@ -426,21 +438,25 @@ int bt_mesh_test_send(uint16_t addr, size_t len,
...
 	}
 
-	err = k_sem_take(&sem, timeout);
+	err = k_sem_take(&send_ctx.sem, timeout);
 	if (err) {
 		LOG_ERR("Send timed out");
 		return err;
 	}
 
+	if (send_ctx.err) {
```
[the remark this answers is not in the document]
Comment (PavelVPV) [reply tests/bluetooth/bsim_bt/bsim_test_mesh/src/mesh_test.c:456]: This is already logged in `end_tx`.

**Review by alxelax — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]
LGTM 👍

**Review by trond-snekvik — APPROVED** [review]

**Pull request merged**